app/assets/javascripts/trackset.js

```diff
@@ -1,294 +1,80 @@
 // represents a set of patterns to be displayed
 function TrackSet(count){
-	this.maxID = 1; // take this out when rhombus provides a way for instances of a pattern in a track to be given an ID
-	this.host = undefined;
-	this.currentPattern = undefined;
-	this.previousPattern = undefined;
 	this.selectedSet = new Array();
 	this.selectedCount = 0;
-	this.tracks = new Array();
-	this.rtracks = new Array();
-	for(var i = 0; i < count; i++){
-		this.tracks[i] = new SortedList();
-	}
-}
-
-TrackSet.prototype.AddTrack = function(track_object, index){
-	// add a new track to the trackset
-	var count = this.tracks.length;
-	this.tracks[count] = new SortedList();
-	this.rtracks[count] = parseInt(index);
 }
 
-TrackSet.prototype.RemoveTrack = function(track_object){
-	for(var i = this.rtracks.indexOf(parseInt(track_object.index)); i < (this.tracks.length - 1); i++){
-		this.tracks[i] = this.tracks[i+1];
-		this.rtracks[i] = this.rtracks[i+1];
-		var track = this.tracks[i].toArray();
-		for(var index in track){
-			track[index].trackIndex--;
-		}
-	}
-	this.tracks.splice(this.tracks.length-1, 1);
-	this.rtracks.splice(this.rtracks.length-1, 1);
+TrackSet.prototype.AddTrack = function(){
+	rhomb._song.addTrack();

```

Comment:
addTrack() is being called here and in tracklist.js, so two tracks are added instead of one. Commenting out this call seems to make it work.